pkg/apis/core/validation/validation.go

```diff
@@ -3075,6 +3075,32 @@ func validateContainerCommon(ctr *core.Container, volumes map[string]core.Volume
 	allErrs = append(allErrs, validatePullPolicy(ctr.ImagePullPolicy, path.Child("imagePullPolicy"))...)
 	allErrs = append(allErrs, ValidateResourceRequirements(&ctr.Resources, path.Child("resources"), opts)...)
 	allErrs = append(allErrs, ValidateSecurityContext(ctr.SecurityContext, path.Child("securityContext"))...)
+	return allErrs
+}
+
+func validateHostUsers(spec *core.PodSpec, volumes []core.Volume, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Only make the following checks if hostUsers is false (otherwise, the container uses the
+	// same userns as the host, and so there isn't anything to check).
+	if spec.SecurityContext == nil || spec.SecurityContext.HostUsers == nil || *spec.SecurityContext.HostUsers == true {
+		return allErrs
+	}
+
+	// For now only these volumes are supported:
+	// - configmap
+	// - secret
+	// - downwardAPI
+	// - emptyDir
+	// - projected
+	// So reject anything else.
+	for i, vol := range volumes {
+		if vol.EmptyDir != nil || vol.Secret != nil || vol.DownwardAPI != nil || vol.ConfigMap != nil ||
```

Comment:
nite, but maybe easier to read:

```
switch {
case vol.EmptyDir != nil:
case vol.Secret != nil:
case vol.DownwardAPI != nil:
case vol.ConfigMap != nil:
case vol.Projected != nil:
default:
	allErrs = append(allErrs, field.Forbidden(fldPath.Child("volumes").Index(i), "volume type not yet supported in combination with pod.Spec.HostUsers=false"))
```

Reply:
Also you want to say:
```
field.Forbidden(fldPath.Child("volumes").Index(i), "volume type not yet supported in combination with `pod.spec.hostUsers`=false"))
```

note the backticks

Reply:
I'm not sure forbidden is right here, but NotSupported is awkward, too.  Meh..

Reply:
@thockin thanks! That looks nicer, adding it now that I'm changing the validation function.

re backticks: I was not aware we were using them, thanks. I changed it a bit to fit with style I found in the file using backticks: removed the "=false" and used "is false".

re forbidden: It took me a while to decide that one. But the documentation for NotSupported was clearly saying it should not be used in this case (see [here](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation/field#NotSupported): `This is used to report unknown values for enumerated fields (e.g. a list of valid values).`). And other code in validation was using forbidden for similar validations. So, I ended up using that.